utils/clustering-functions.R

```diff
@@ -62,8 +62,42 @@ kmeans_clustering <- function(normalized_sce,
   
 }
 
+define_nn_range <- function(nearest_neighbors_lowest,
+                            nearest_neighbors_highest,
+                            nearest_neighbors_increment = NULL){
+  # Purpose: Define a nearest neighbors range sequence with the provided
+  # range and increment values
+  
+  # Args:
+  #   nearest_neighbors_lowest: lowest number of a range of nearest neighbors 
+  #                             values to include when calculating/plotting the 
+  #                             clustering results.
+  #   nearest_neighbors_highest: highest number of a range of nearest neighbors 
+  #                             values to include when calculating/plotting the 
+  #                             clustering results.
+  #   nearest_neighbors_increment: increment to use when implementing the range 
+  #                                number of nearest neighbors for cluster stats.
+  
+  # Define range using the lowest and highest values
+  nearest_neighbors_range <- c(nearest_neighbors_lowest:nearest_neighbors_highest)
+  
+  # If no nearest neighbors increment has been input then the provided range is 
+  # directly used for clustering 
+  if (is.null(nearest_neighbors_increment)) {
+    nn_range <- nearest_neighbors_range
+  } else {
+    # If a nearest neighbors increment exists, then create a sequence for clustering 
+    nn_range <- seq(min(nearest_neighbors_range), 
+                    max(nearest_neighbors_range), 
+                    nearest_neighbors_increment) 
+  }
```

Comment:
```suggestion
  if (is.null(nearest_neighbors_increment)) {
    nearest_neighbors_increment <-1
  } 
  nn_range <- seq(min(nearest_neighbors_range), 
                  max(nearest_neighbors_range), 
                  nearest_neighbors_increment) 

```
So I think here you just need to assign the increment to something and then create the range regardless. I'm not sure what you had will work if the increment was NULL.

Reply:
Ah yes, I think this is it -- this makes sense (will still test for the various possible cases here) 👍